src/envoy/tcp/alpn_proxy/config/alpn_proxy.proto

```diff
@@ -0,0 +1,46 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package envoy.tcp.alpnproxy.config;
+
+option java_outer_classname = "AlpnProxyProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.tcp.alpnproxy.config";
+option go_package = "AlpnProxy";
+
+// [#protodoc-title: AlpnProxy protocol match and data transfer]
+// AlpnProxy protocol match and data transfer
+message AlpnProxy {
+  // Direction of the flow of traffic in which this this AlpnProxy filter
+  // is plcaed.
+  enum FilterDirection {
+    DOWNSTREAM = 0;
+    UPSTREAM = 1;
+  }
+  // Protocol that Alpn should support on the server.
+  // [#comment:TODO(GargNupur): Make it a list.]
+  string protocol = 1;
+
+  // The node metadata id whose data will be written to connection data.
+  string node_metadata_id = 2;
+
+  // The human readable prefix to use when emitting
+  string stat_prefix = 3;
+
+  // Defines if this is a downstream or upstream AlpnProxy filter.
+  FilterDirection filter_direction = 4;
```

Comment:
Is this implicit in the distinction between network and upstream network filters? It'd be better to have less config since the direction should be inferrable from the factory.

Reply:
Thanks.. Fixed